File: crates/codegraph-extract/tests/fixtures/sample.js
import path from "path";

export function processUser(id) {
  return formatEmail(id);
}

function formatEmail(s) {
  return s.toLowerCase();
}

export class UserService {
  constructor(name) { this.name = name; }
  greet() { return processUser(this.name); }
}

File: crates/codegraph-extract/tests/fixtures/sample.rs
use std::collections::HashMap;

pub fn process_user(id: &str) -> String {
    format_email(id)
}

fn format_email(s: &str) -> String {
    s.to_lowercase()
}

pub struct UserService {
    pub name: String,
}

impl UserService {
    pub fn greet(&self) -> String {
        process_user(&self.name)
    }
}
